applications/MeshMovingApplication/custom_utilities/calculate_mesh_velocity_utility.cpp

```diff
@@ -0,0 +1,203 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//                   Andreas Winterstein
+//
+
+
+// System includes
+#include<map>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/mesh_moving_variables.h"
+#include "calculate_mesh_velocity_utility.h"
+
+
+namespace Kratos
+{
+
+CalculateMeshVelocityUtility::CalculateMeshVelocityUtility(ModelPart& rModelPart,
+                                                           Parameters Settings)
+    : mrModelPart(rModelPart)
+{
+    KRATOS_ERROR_IF_NOT(rModelPart.HasNodalSolutionStepVariable(MESH_DISPLACEMENT))
+        << "The ModelPart does not have the variable \"MESH_DISPLACEMENT\" as "
+        << "nodal-solutionstepvariable!" << std::endl;
+    KRATOS_ERROR_IF_NOT(rModelPart.HasNodalSolutionStepVariable(MESH_VELOCITY))
+        << "The ModelPart does not have the variable \"MESH_VELOCITY\" as "
+        << "nodal-solutionstepvariable!" << std::endl;
+
+    const std::string time_scheme = Settings["time_scheme"].GetString();
+
+    mIntegrationMethod = std::get<0>(GetMethodIterator(time_scheme)->second);
+
+    const SizeType min_buffer_size = GetMinimumBufferSize(time_scheme);
+    const SizeType current_buffer_size = rModelPart.GetBufferSize();
+
+    KRATOS_ERROR_IF(current_buffer_size < min_buffer_size) << "Insufficient buffer size: "
+        << current_buffer_size << " < " << min_buffer_size << "!" << std::endl;
+
+    // set Generalized-Alpha specific Settings
+    if (mIntegrationMethod == generalized_alpha) {
+        KRATOS_ERROR_IF_NOT(rModelPart.HasNodalSolutionStepVariable(MESH_ACCELERATION))
+            << "The ModelPart does not have the variable \"MESH_ACCELERATION\" as "
+            << "nodal-solutionstepvariable!" << std::endl;
+
+        // default (Newmark)
+        double alpha_m = 0.0;
+        double alpha_f = 0.0;
+
+        // setting mAlphaF and mAlphaM
+        if (time_scheme == "generalized_alpha") {
+            alpha_m = Settings["alpha_m"].GetDouble();
+            alpha_f = Settings["alpha_f"].GetDouble();
+        } else if (time_scheme == "bossak") {
+            if (Settings.Has("alpha_m")) {
+                alpha_m = Settings["alpha_m"].GetDouble();
+            }
+            else {
+                alpha_m = -0.3; // default in Kratos
+            }
+        }
+
+        mBossakBeta = std::pow((1.0 + alpha_f - alpha_m), 2) * 0.25;
```

Comment:
instead of pow (x,2) , can use x*x
I remember reading pow is slow.

Reply:
As a comment aside, I think that `pow(x,2)` is faster (or at least equivalent) to `x*x`when the exponent is an integer.